core/lib/state/src/catchup.rs

```diff
@@ -313,23 +313,32 @@ mod tests {
         let keep_updated_task = rocksdb_cell.keep_updated(pool.clone());
         let (_stop_sender, stop_receiver) = watch::channel(false);
         let catchup_task_handle = tokio::spawn(catchup_task.run(stop_receiver.clone()));
-        let _keep_updated_task_handle = tokio::spawn(keep_updated_task.run(stop_receiver));
+        let keep_updated_task_handle = tokio::spawn(keep_updated_task.run(stop_receiver));
         catchup_task_handle.await.unwrap().unwrap();
 
         let storage_logs = gen_storage_logs(40..50);
         let mut conn = pool.connection().await.unwrap();
         create_l2_block(&mut conn, L2BlockNumber(2), storage_logs.clone()).await;
-        create_l1_batch(&mut conn, L1BatchNumber(2), &storage_logs).await;
+        // Batch info should be inserted atomically.
+        let mut transaction = conn.start_transaction().await.unwrap();
```

Comment:
Nit: I'd cover the preceding `create_l2_block()` by the same transaction as well.